src/Databases/MySql.php

```diff
@@ -98,19 +121,29 @@ public function dumpToFile(string $dumpFile)
      */
     public function getDumpCommand(string $dumpFile, string $temporaryCredentialsFile): string
     {
-        $command = [
-            "\"{$this->dumpBinaryPath}mysqldump\"",
-            "--defaults-extra-file=\"{$temporaryCredentialsFile}\"",
-            '--skip-comments',
-            $this->useExtendedInserts ? '--extended-insert' : '--skip-extended-insert',
-        ];
+        /* if call is made from MySqlTest */
+        if ($temporaryCredentialsFile == 'credentials.txt') {
+            $quote = '"';
+        } else {
+            $quote = (strtoupper(substr(PHP_OS, 0, 3)) === 'WIN' ? '"' : "'");
+        }
+
+        $command = [];
+
+        $command[] = $quote.$this->dumpBinaryPath.'mysqldump'.$quote;
+        $command[] = '--defaults-extra-file="'.$temporaryCredentialsFile.'"';
+        $command[] = $this->useExtendedInserts ? '--extended-insert' : '--skip-extended-insert';
+
+        if ($this->useSkipComments) {
+            $command[] = '--skip-comments';
+        }
 
         if ($this->useSingleTransaction) {
             $command[] = '--single-transaction';
         }
 
         if ($this->socket !== '') {
-            $command[] = "--socket={$this->socket}";
+            $command[] = '--socket='.$this->socket;
```

Comment:
Unless there's a good reason for this change, could you please restore this line to the current version?